AndelaOSP/shufflebox-backend
# Refactor hangout model to have groups

#### What does this PR do?
**Proposes changes to the Hangout model as follows**:
* Refactor the hangout model to have groups that contain members instead of directly having members in itself.
* Disable direct creation of hangouts in favour of creating hangouts during a shuffle.
* Correct tests to adapt to new model.
* Limit hangouts to one per month, dated the last.

@gitgik will have a look at this proposal implementation and let the team know if this makes any sense.

#### Description of Task to be completed?
#### How should this be manually tested?
If you already have a database for the application setup, you'll need to delete and recreate it then rerun the migrations. Visit `/api/users/` to create users that'll be be grouped when shuffling a hangout, then using a Rest client, e.g. Postman, issue a post request to `/api/shuffle/` with data such as
```json
{ "type": "hangout", "limit": 3}
```
Any number can be the limit. The default is 10. You should get back a single hangout with a single or multiple groups depending on the number of your users and `limit` value. Issuing the same request again should yield
```json
{
  "message": "Hangout groups for this month were already created."
}
```

#### Any background context you want to provide?
Previously, shuffling hangouts would result to multiple hangouts with the same date but different members. This PR proposes to change that such that shuffling creates a single, monthly hangout. Each such hangout will have multiple groups and every group will have multiple users.

#### What are the relevant pivotal tracker stories?
#### Screenshots (if appropriate)
#### Questions:
